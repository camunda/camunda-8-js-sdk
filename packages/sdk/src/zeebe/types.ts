export * from './lib/interfaces-1.0'
export * from './lib/interfaces-grpc-1.0'
export * from './lib/interfaces-published-contract'
