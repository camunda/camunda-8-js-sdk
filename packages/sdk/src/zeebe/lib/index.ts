export * from './BpmnParser'
export * from './stringifyVariables'
