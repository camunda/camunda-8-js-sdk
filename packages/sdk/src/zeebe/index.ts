export { Duration } from 'typed-duration'
export { BpmnParser } from './lib/BpmnParser'
import { ZBSimpleLogger } from './lib/SimpleLogger'
import { ZBJsonLogger } from './lib/ZBJsonLogger'
import { ZBLogger } from './lib/ZBLogger'
import * as Interfaces from './lib/interfaces-1.0'
import * as GrpcInterfaces from './lib/interfaces-grpc-1.0'
import * as PublishedContract from './lib/interfaces-published-contract'

export * from './zb/ZBWorker'
export * from './zb/ZeebeGrpcClient'
export * from './zb/ZeebeRESTClient'
/** @namespace */
export const Types = {
	...Interfaces,
	...GrpcInterfaces,
	...PublishedContract,
}

export const Logging = {
	ZBJsonLogger,
	ZBLogger,
	ZBSimpleLogger,
}
