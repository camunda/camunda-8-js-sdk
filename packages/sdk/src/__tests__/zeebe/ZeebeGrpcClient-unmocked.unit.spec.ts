import {
	EnvironmentSetup,
	restoreZeebeLogging,
	suppressZeebeLogging,
} from '../../lib'
import { ZeebeGrpcClient } from '../../zeebe'

beforeAll(() => EnvironmentSetup.storeEnv())
beforeEach(() => {
	EnvironmentSetup.wipeEnv()
	suppressZeebeLogging()
})
afterAll(() => {
	restoreZeebeLogging()
	EnvironmentSetup.restoreEnv()
})

jest.setTimeout(13000)

test('ZeebeGrpcClient constructor throws an exception when there is no broker and retry is false', async () => {
	const zbc = new ZeebeGrpcClient({
		config: {
			CAMUNDA_OAUTH_DISABLED: true,
			ZEEBE_ADDRESS: 'localhoster',
			zeebeGrpcSettings: {
				ZEEBE_GRPC_CLIENT_RETRY: false,
			},
		},
	})
	expect.assertions(1)
	try {
		await zbc.deployResource({
			processFilename: './src/__tests__/testdata/hello-world.bpmn',
		})
	} catch (e: unknown) {
		expect((e as Error).message.indexOf('14 UNAVAILABLE:')).toEqual(0)
	}
	await zbc.close()
})

test('cancelProcessInstance throws an exception when workflowInstanceKey is malformed', async () => {
	const zbc = new ZeebeGrpcClient({
		config: {
			CAMUNDA_OAUTH_DISABLED: true,
			ZEEBE_ADDRESS: 'localhoster',
			zeebeGrpcSettings: { ZEEBE_GRPC_CLIENT_RETRY: false },
		},
	})
	expect.assertions(1)
	try {
		await zbc.cancelProcessInstance('hello-world')
	} catch (e: unknown) {
		expect(e).toMatchSnapshot()
	}
	await zbc.close()
})
