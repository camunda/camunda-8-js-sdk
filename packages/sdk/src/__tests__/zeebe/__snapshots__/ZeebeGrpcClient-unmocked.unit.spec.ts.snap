// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cancelProcessInstance throws an exception when workflowInstanceKey is malformed 1`] = `
[Error: 
		  processInstanceKey is malformed, value : hello-world
		  ]
`;
